tools/segmetrics/run-segmetrics.py

```diff
@@ -1,133 +1,59 @@
 """
-Copyright 2022 Leonid Kostrykin, Biomedical Computer Vision Group, Heidelberg University.
+Copyright 2022-2023 Leonid Kostrykin, Biomedical Computer Vision Group, Heidelberg University.
 
 Distributed under the MIT license.
 See file LICENSE for detail or copy at https://opensource.org/licenses/MIT
 
 """
 
 import argparse
-import csv
-import itertools
 import pathlib
+import subprocess
 import tempfile
 import zipfile
 
-import numpy as np
-import segmetrics as sm
-import skimage.io
+import pandas as pd
 
 
-measures = [
-    ('dice', 'Dice', sm.regional.Dice()),
-    ('seg', 'SEG', sm.regional.ISBIScore()),
-    ('jc', 'Jaccard coefficient', sm.regional.JaccardSimilarityIndex()),
-    ('ji', 'Jaccard index', sm.regional.JaccardIndex()),
-    ('ri', 'Rand index', sm.regional.RandIndex()),
-    ('ari', 'Adjusted Rand index', sm.regional.AdjustedRandIndex()),
-    ('hsd_sym', 'HSD (sym)', sm.boundary.Hausdorff('sym')),
-    ('hsd_e2a', 'HSD (e2a)', sm.boundary.Hausdorff('e2a')),
-    ('hsd_a2e', 'HSD (a2e)', sm.boundary.Hausdorff('a2e')),
-    ('nsd', 'NSD', sm.boundary.NSD()),
-    ('o_hsd_sym', 'Ob. HSD (sym)', sm.boundary.ObjectBasedDistance(sm.boundary.Hausdorff('sym'))),
-    ('o_hsd_e2a', 'Ob. HSD (e2a)', sm.boundary.ObjectBasedDistance(sm.boundary.Hausdorff('e2a'))),
-    ('o_hsd_a2e', 'Ob. HSD (a2e)', sm.boundary.ObjectBasedDistance(sm.boundary.Hausdorff('a2e'))),
-    ('o_nsd', 'Ob. NSD', sm.boundary.ObjectBasedDistance(sm.boundary.NSD())),
-    ('fs', 'Split', sm.detection.FalseSplit()),
-    ('fm', 'Merge', sm.detection.FalseMerge()),
-    ('fp', 'Spurious', sm.detection.FalsePositive()),
-    ('fn', 'Missing', sm.detection.FalseNegative()),
-]
-
-
-def process_batch(study, gt_filelist, seg_filelist, namelist, gt_is_unique, seg_is_unique):
-    for gt_filename, seg_filename, name in zip(gt_filelist, seg_filelist, namelist):
-        img_ref = skimage.io.imread(gt_filename)
-        img_seg = skimage.io.imread(seg_filename)
-        study.set_expected(img_ref, unique=gt_is_unique)
-        study.process(img_seg, unique=seg_is_unique, chunk_id=name)
-
-
-def aggregate(measure, values):
-    fnc = np.sum if measure.ACCUMULATIVE else np.mean
-    return fnc(values)
-
-
-def is_zip_filepath(filepath):
-    return filepath.lower().endswith('.zip')
-
-
-def is_image_filepath(filepath):
-    suffixes = ['png', 'tif', 'tiff']
-    return any((filepath.lower().endswith(f'.{suffix}') for suffix in suffixes))
+def process_batch(seg_dir, seg_file, gt_file, tsv_output_file, recursive, gt_unique, seg_unique, measures):
+    with tempfile.NamedTemporaryFile() as csv_output_file:
+        cmd = ['python', '-m', 'segmetrics.cli', str(seg_dir), str(seg_file), str(gt_file), str(csv_output_file.name), '--semicolon']
+        if recursive:
+            cmd.append('--recursive')
+        if gt_unique:
+            cmd.append('--gt-unique')
+        if seg_unique:
+            cmd.append('--seg-unique')
+        cmd += measures
+        print(cmd)

```

Comment:
this can be removed?

Reply:
Thanks! I forgot to remove some code temporarily used for debugging… will fix that.